packages/docs/src/pages/guides/typescript.mdx

```diff
@@ -10,56 +10,54 @@ While most APIs in Theme UI should _just work_ in TypeScript,
 there are a few advanced use cases which will differ slightly.
 This guide is intended to cover those use cases.
 
-## Exact theme type
+## Extending the Theme
 
-The [`Theme` type](/theme-spec) represents all possible themes.
+If you need additional properties in the theme object,
+you can add them with [module augmentation](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation).
 
-It might be what you need when you're writing a library of reusable components
-or an app where the theme object is provided by the user and kept in the database.
-However, it's not the most convenient way to think about the theme,
-when building a blog or a landing page.
+```tsx
+import { makeTheme, UserTheme } from '@theme-ui/css';
 
-To know the exact type of your particular theme on type level,
-you can use an identity "constructor" function to narrow the type.
+interface MyTheme {
+  colors: {
+    primary: string;
+    'primary-dark': string;
+    'primary-light': string;
+  }
+}
 
-```tsx
-const makeTheme = <T extends Theme>(t: T) => t
+declare module '@theme-ui/css' {
+  interface UserTheme extends MyTheme {}
+}
 
 const theme = makeTheme({
   colors: {
-    background: 'white',
-    text: 'black',
-    blue: {
-      light: '#187abf',
-      dark: '#235a97',
-    },
+    primary: 'red',
+    'primary-dark': 'darkred',
+    'primary-light': 'pink'
   },
 })
-
-export type ExactTheme = typeof theme
-
-// No error
-const lightBlue = theme.colors.blue.light
 ```
 
-You can then reexport `useThemeUI` hook with narrowed type.
+theme-ui's `Theme` interface will automatically pick up customizations to
+`UserTheme` so you will be free to reference your custom properties:
 
 ```tsx
-import { ContextValue } from 'theme-ui'
-
-interface ExactContextValue extends Omit<ContextValue, 'theme'> {
-  theme: ExactTheme
-}
+import { css } from '@theme-ui/css';
 
-export const useTheme = (useThemeUI as unknown) as () => ExactContextValue
-```
+// using css function
+const primaryDarkColor = css({
+  color: 'primary-dark' // auto-completes and no error
+})
 
-_[Try it on CodeSandbox.](https://codesandbox.io/s/theme-ui-typescript-tips-yh8u1?file=/src/exact-theme-type.ts)_
+// using jsx pragma
```

Comment:
Everywhere we use the JSX pragma, should include the `/** jsx` line

Reply:
I will add `/** @jsxImportSource theme-ui */` for consistency.

Reply:
Also using jsxImportSource in the new TS example I added to this PR!